phylanx/execution_tree/primitives/base_primitive.hpp

```diff
@@ -169,9 +169,7 @@ namespace phylanx { namespace execution_tree
     ///////////////////////////////////////////////////////////////////////////
     // Factory functions
     using factory_function_type =
-        hpx::util::function_nonser<
-            primitive(hpx::id_type, std::vector<primitive_argument_type>&&)
-        >;
+        primitive (*)(hpx::id_type, std::vector<primitive_argument_type>&&);
```

Comment:
What does this change do?

Reply:
It was a `function<>` object before and is a plain pointer to a (C-style) function now. This was a workaround for a problem in HPX which hopefully will be solved soon (see https://github.com/STEllAR-GROUP/hpx/pull/2945), at which point we can roll back this change.